ext/ddtrace_profiling_native_extension/extconf.rb

```diff
@@ -0,0 +1,14 @@
+require 'mkmf'
+
+# We don't support JRuby for profiling, and JRuby doesn't support native extensions, so let's just skip this entire
+# thing so that JRuby users of dd-trace-rb aren't impacted.
+if RUBY_ENGINE == 'jruby'
+  File.write('Makefile', dummy_makefile($srcdir).join) # rubocop:disable Style/GlobalVars
+  return
+end
+
+# Tag the native extension library with the Ruby version and Ruby platform.
+# This makes it easier for development (avoids "oops I forgot to rebuild when I switched my Ruby") and ensures that
+# the wrong library is never loaded.
+# When requiring, we need to use the exact same string, including the version and the platform.
+create_makefile "ddtrace_profiling_native_extension.#{RUBY_VERSION}_#{RUBY_PLATFORM}"
```

Comment:
Do you think it makes sense to also add the OS, given we use MacOS+Docker for development in the same shared folder?

Reply:
`RUBY_PLATFORM` already includes the OS:
* `x86_64-darwin19` this is what I get on my dev machine
* `x86_64-linux-musl` this is docker + alpine linux
* `x86_64-linux` this is docker + regular linux

(Also the compiled library is going to have a `.bundle` extension on macos, while linux uses `.so`, so no collision there).

The one thing not included here is `RUBY_ENGINE`, which may or may not be relevant for TruffleRuby but... I'm not sure it's worth it.